docs(oss): add langchain/production ui docs (#367)
langchain-ai/docs: src/oss/langchain-ui.mdx

---
title: Prebuilt UI
---

import AlphaCallout from '/snippets/alpha-lc-callout.mdx';

<AlphaCallout />

> LangChain provides powerful prebuilt user interfaces that work seamlessly with agents created using `create_agent`. These UIs are designed to provide rich, interactive experiences for your agents with minimal setup, whether you're running locally or deployed on LangGraph Platform.

Reviewer: ```suggestion
LangChain provides powerful prebuilt user interfaces that work seamlessly with agents created using `create_agent`. These UIs are designed to provide rich, interactive experiences for your agents with minimal setup, whether you're running locally or in a deployed context (such as LangGraph Platform).
```

Would also want to link "LangGraph Platform" to product page

## Overview

Our prebuilt UIs offer two complementary interfaces for different use cases:

- **[Agent Chat UI](#agent-chat-ui)**: A conversational interface for real-time chat interactions with your agents, featuring tool visualization and generative UI capabilities
- **[Agent Inbox](#agent-inbox)**: A task management interface specifically designed for human-in-the-loop workflows, allowing users to review, approve, and respond to agent requests

Both UIs are open source and can be adapated to your application needs.

## Agent Chat UI

> Agent Chat UI is a Next.js application that provides a conversational interface for interacting with any LangGraph agent. It supports real-time chat, tool visualization, and advanced LangGraph features like time-travel debugging and state forking.

Reviewer 2: mentions of langgraph... should just be langchian?

> Agent Chat UI is a Next.js application that provides a conversational interface for interacting with any LangGraph agent. It supports real-time chat, tool visualization, and advanced LangGraph features like time-travel debugging and state forking.

Reviewer: ```suggestion
Agent Chat UI is a Next.js application that provides a conversational interface for interacting with any LangChain agent. It supports real-time chat, tool visualization, and advanced LangGraph features like time-travel debugging and state forking.
```

2. **Deployment URL**: Your LangGraph server's endpoint (e.g., `http://localhost:2024` for local development, or your deployed agent's URL)
3. **LangSmith API key (optional)**: Add your LangSmith API key (not required if you're using a local LangGraph server)

Once configured, Agent Chat UI will automatically fetch and display any interrupted threads from your agent.

<Tip>
  Agent Chat UI has out-of-the-box support for rendering tool calls and tool result messages. To customize what messages are shown, see the [Hiding Messages in the Chat](https://github.com/langchain-ai/agent-chat-ui?tab=readme-ov-file#hiding-messages-in-the-chat) section in the Agent Chat UI documentation.
</Tip>

### Generative UI

<Frame>
![Agent view in LangGraph studio UI](../images/gen_ui.gif)
</Frame>

Agent Chat UI supports generative UI capabilities, allowing your agents to dynamically create rich, interactive components based on the conversation context. This enables more engaging user experiences beyond simple text responses.

Some examples of what you can build with generative UI:
- Create interactive forms and input components
- Generate charts, graphs, and creative data visualizations

> - Render rich media content dynamically, for example, videos

Reviewer: ```suggestion
- Render rich media content dynamically (e.g. images and videos)
```
